### PR TITLE
Revert if ETH return is unsuccessful

### PR DESCRIPTION
**Pull request by alicevhenshaw**
## Description

https://github.com/airswap/airswap-protocols/issues/298

Quick description:

- [ ] N/A
- [ ] New features
- [X] Bug fixes
- [ ] Typo/small tweaks

## Changes

- Added a revert statement when ETH fails to send to a non-payable contract in the wrapper
- Created a mock contract that is non-payable to test this new revert
- Added an integration test that uses the new mock to test the revert

## Tests
```
Wrapper:
  39 passing (10s)
```

### CODE REVIEW

File: source/wrapper/contracts/Wrapper.sol [hunk 1]
```diff
@@ -148,7 +148,9 @@ contract Wrapper is Ownable {
 
       // Transfer ether to the user.
       // solium-disable-next-line security/no-call-value
```
Comment (poiresel) [inline source/wrapper/contracts/Wrapper.sol]: can remove the  // solium-disable-next-line security/no-call-value now

Comment (alicevhenshaw) [reply source/wrapper/contracts/Wrapper.sol]: It's still doing a call value, and so is still considered a security flaw as you pass the recipient contract access to all gas and allow reentrancy. Solium will still warn of this even if we catch the bool as you can do reentrancy as then succeed.

Comment (poiresel) [reply source/wrapper/contracts/Wrapper.sol]: Interesting, ok.

File: source/wrapper/contracts/HelperMock.sol [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  function authorizeWrapperToSend() external {
+    ISwap swap = wrapper.swapContract();
+    swap.authorizeSender(address(wrapper));
+  }
+
+  function approveToken(IERC20 token, address toApprove, uint256 amount) external {
+    token.approve(toApprove, amount);
+  }
+
+}
```
Comment (ejwessel) [inline source/wrapper/contracts/HelperMock.sol]: Can we add a note somewhere in here to denote its use is exclusively for non payable?

Comment (alicevhenshaw) [reply source/wrapper/contracts/HelperMock.sol]: Yeah nice, I'll add a natspec at the top of the contract to explain why its been created.

**Pull request merged**